### PR TITLE
Move remaining XDS vertical API calls to use kubecontroller caches

### PR DESCRIPTION
**Pull request by eduser25**
GetPodFromCertificate and ListPods were two remaining APIs used by
pretty much all verticals which were still going through kubeclient API
REST calls.

In E2E tests, with a higher count of pods we've seen slowness issues,
which ranged k8s API throttling, go-routine contention (as envoy updates pile up)
and webhooks timeout.

These additions remove any network/API dependency from the XDS paths, making
verticals' execution range between 50-100ms in demo conditions and debug logging.
(Down from 7-10seconds per vertical)

Please describe the motivation for this PR and provide enough
information so that others can review it.

- Control Plane          [X]
- Tests / CI System      [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

**Comment by eduser25** [comment]
Additionally, bookbuyer in demo should start seeing success after 6-10 iterations, down from 30

File: pkg/catalog/fake.go [hunk 1]
```diff
@@ -52,23 +52,17 @@ func NewFakeMeshCatalog(kubeClient kubernetes.Interface) *MeshCatalog {
...
 		// play pretend this call queries a controller cache
 		var services []*corev1.Service
 
-		for _, ns := range listExpectedNs {
-			svcList, _ := kubeClient.CoreV1().Services(ns).List(context.Background(), metav1.ListOptions{})
-			for _, svcItem := range svcList.Items {
-				services = append(services, &svcItem)
-			}
+		// Things worked before because internally, the APIs were calling kubeClient APIs with "namespace"
+		// parametre. Some tests have been using catalog APIs bypassing the namespace monitoring altogether.
```
Comment (shashankram) [inline pkg/catalog/fake.go:61]: nit
```suggestion
		// parameter. Some tests have been using catalog APIs bypassing the namespace monitoring altogether.
```

File: pkg/catalog/fake.go [hunk 2]
```diff
@@ -52,23 +52,17 @@ func NewFakeMeshCatalog(kubeClient kubernetes.Interface) *MeshCatalog {
...
 	mockKubeController.EXPECT().ListServices().DoAndReturn(func() []*corev1.Service {
 		// play pretend this call queries a controller cache
 		var services []*corev1.Service
 
-		for _, ns := range listExpectedNs {
-			svcList, _ := kubeClient.CoreV1().Services(ns).List(context.Background(), metav1.ListOptions{})
-			for _, svcItem := range svcList.Items {
-				services = append(services, &svcItem)
-			}
+		// Things worked before because internally, the APIs were calling kubeClient APIs with "namespace"
```
Comment (shashankram) [inline pkg/catalog/fake.go:60]: This comment is not very helpful. Once this is merged `Things worked before because...` lacks context.

File: pkg/catalog/fake.go [hunk 3]
```diff
@@ -82,6 +76,20 @@ func NewFakeMeshCatalog(kubeClient kubernetes.Interface) *MeshCatalog {
...
+		if err != nil {
+			return nil
+		}
+
+		var podRet []*v1.Pod = []*v1.Pod{}
+		for idx := range vv.Items {
+			podRet = append(podRet, &vv.Items[idx])
+		}
+		return podRet
+	}).AnyTimes()
```
Comment (shashankram) [inline pkg/catalog/fake.go:91]: Why is this needed?

**Review by shashankram — DISMISSED** [review]

**Comment by shashankram** [comment]
> Additionally, bookbuyer in demo should start seeing success after 6-10 iterations, down from 30

That's a nice improvement.

File: pkg/catalog/fake.go (hunk 2 above)
Comment (eduser25) [reply pkg/catalog/fake.go:60]: I left it for review purposes mostly. I imagine that if I stayed with the "assumption" part, reviews might have asked "why change it".

File: pkg/catalog/fake.go (hunk 3 above)
Comment (eduser25) [reply pkg/catalog/fake.go:91]: This is needed for the tests run catalog functions, as I can't interface with the kubecontroller inside it. I changed some tests to just run a separate kubecontroller as they just test xds_cert functions, and they don't need catalog.
But for the ones running catalog, they need this.

File: pkg/catalog/fake.go (hunk 2 above)
Comment (eduser25) [reply pkg/catalog/fake.go:60]: will change it to just leave the assumption part, should be explanatory enough

File: pkg/catalog/fake.go (hunk 3 above)
Comment (shashankram) [reply pkg/catalog/fake.go:91]: I'd drop a comment here and file an issue to track such custom fake code needing enhancements.

Comment (eduser25) [reply pkg/catalog/fake.go:91]: +1, could we maybe reuse https://github.com/openservicemesh/osm/issues/1683 and generalize the issue?

File: pkg/catalog/fake.go (hunk 1 above)
Comment (eduser25) [reply pkg/catalog/fake.go]: done

**Review by shashankram — APPROVED** [review]

**Review by nojnhuh — APPROVED** [review]
